examples/cpu/par_example.cpp

```diff
@@ -0,0 +1,158 @@
+                                    "rot_zw"});
+    auto surface_transforms = traccc::read_surfaces(sreader);
+
+    // Algorithms
+    traccc::component_connection cc;
+    traccc::measurement_creation mt;
+    traccc::spacepoint_formation sp;
+
+    // Output stats
+    uint64_t n_cells = 0;
+    uint64_t m_modules = 0;
+    uint64_t n_clusters = 0;
+    uint64_t n_measurements = 0;
+    uint64_t n_space_points = 0;
+
+    // Memory resource used by the EDM.
+    vecmem::host_memory_resource resource;
+
+#pragma omp parallel for reduction (+:n_cells, n_clusters, n_measurements, n_space_points, m_modules)
+    // Loop over events
+    for (unsigned int event = 0; event < events; ++event) {
+
+        // Read the cells from the relevant event file
+        std::string event_string = "000000000";
+        std::string event_number = std::to_string(event);
+        event_string.replace(event_string.size() - event_number.size(), event_number.size(), event_number);
+
+        std::string io_cells_file =
+                data_directory + cells_dir + std::string("/event") + event_string + std::string("-cells.csv");
+        traccc::cell_reader creader(io_cells_file,
+                                    {"geometry_id", "hit_id", "cannel0", "channel1", "activation", "time"});
+        traccc::host_cell_container cells_per_event = traccc::read_cells(creader, resource, surface_transforms);
+        m_modules += cells_per_event.headers.size();
+
+        // Output containers
+        traccc::host_measurement_container measurements_per_event;
+        traccc::host_spacepoint_container spacepoints_per_event;
+        measurements_per_event.headers.reserve(cells_per_event.headers.size());
+        measurements_per_event.items.reserve(cells_per_event.headers.size());
+        spacepoints_per_event.headers.reserve(cells_per_event.headers.size());
+        spacepoints_per_event.items.reserve(cells_per_event.headers.size());
+
+#pragma omp parallel for
+        for (std::size_t i = 0; i < cells_per_event.items.size(); ++i) {
+            auto &module = cells_per_event.headers[i];
+            module.pixel = traccc::pixel_segmentation{-8.425, -36.025, 0.05, 0.05};
+
+            // The algorithmic code part: start
+            traccc::cluster_collection clusters_per_module = cc(cells_per_event.items[i], cells_per_event.headers[i]);
+            clusters_per_module.position_from_cell = module.pixel;
+
+            traccc::host_measurement_collection measurements_per_module = mt(clusters_per_module, module);
+            traccc::host_spacepoint_collection spacepoints_per_module = sp(module, measurements_per_module);
+            // The algorithmnic code part: end
+
+            n_cells += cells_per_event.items[i].size();
+            n_clusters += clusters_per_module.items.size();
+            n_measurements += measurements_per_module.size();
+            n_space_points += spacepoints_per_module.size();
+
+#pragma omp critical
```

Comment:
Personally I find it a lot clearer to indent OpenMP pragmas on the same level as the construct they're modifying, but that's up to you.

Reply:
That is also my preference but I chose to format the code (with clang-format using the "llvm" style) before pushing it. Happy to use any style you recommend me so that it is inline with the existing code.